packages/rhi-webgl/src/GLTexture2DArray.ts

```diff
@@ -0,0 +1,111 @@
+import { IPlatformTexture2DArray, Texture2DArray, TextureFormat } from "@oasis-engine/core";
+import { GLTexture } from "./GLTexture";
+import { WebGLRenderer } from "./WebGLRenderer";
+
+/**
+ * Texture 2D array in WebGL platform.
+ */
+export class GLTexture2DArray extends GLTexture implements IPlatformTexture2DArray {
+  constructor(rhi: WebGLRenderer, texture2DArray: Texture2DArray) {
+    super(rhi, texture2DArray, (<WebGL2RenderingContext>rhi.gl).TEXTURE_2D_ARRAY);
+
+    const { format, width, height, length, mipmapCount } = texture2DArray;
+
+    if (!this._isWebGL2) {
+      throw new Error(`Texture2D Array is not supported in WebGL1.0`);
+    }
+
+    if (!GLTexture._supportTextureFormat(format, rhi)) {
+      throw new Error(`Texture format is not supported:${TextureFormat[format]}`);
+    }
+
+    this._formatDetail = GLTexture._getFormatDetail(format, this._gl, true);
+    this._gl.texStorage3D(this._target, mipmapCount, this._formatDetail.internalFormat, width, height, length);
+  }
+
+  /**
+   * {@inheritDoc IPlatformTexture2DArray.setPixelBuffer}
+   */
+  setPixelBuffer(
+    elementIndex: number,
+    colorBuffer: ArrayBufferView,
+    mipLevel: number = 0,
+    x: number = 0,
+    y: number = 0,
+    width?: number,
+    height?: number
+  ): void {
+    const { _target: target, _gl: gl } = this;
+    const { internalFormat, baseFormat, dataType, isCompressed } = this._formatDetail;
+
+    width = width || Math.max(1, this._texture.width >> mipLevel) - x;
+    height = height || Math.max(1, this._texture.height >> mipLevel) - y;
+
+    this._bind();
+    gl.pixelStorei(gl.UNPACK_FLIP_Y_WEBGL, 0);
+    gl.pixelStorei(gl.UNPACK_PREMULTIPLY_ALPHA_WEBGL, 0);
+
+    if (isCompressed) {
+      gl.compressedTexSubImage3D(target, mipLevel, x, y, elementIndex, width, height, 1, internalFormat, colorBuffer);
```

Comment:
should be `gl.compressedTexSubImage3D(target, mipLevel, x, y, 0, width, height, elementIndex, internalFormat, colorBuffer);`